### PR TITLE
Add options for rocksdb

### PR DESCRIPTION
**Pull request by DorianZheng**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -9,15 +9,18 @@ endif
 
 ifeq ($(ROCKSDB_CHECK), 0)
 	TAGS += rocksdb
-endif 
+    CGO_CPPFLAGS += "-std=c++11"
+    CGO_LDFLAGS += "-g -O2 -lrocksdb -lstdc++ -lm -lz -lbz2 -lsnappy -llz4 -lzstd"
```
Comment (siddontang) [inline Makefile]: seem we don't need to pass these here?

File: Makefile [hunk 2]
```diff
@@ -9,15 +9,18 @@ endif
 
 ifeq ($(ROCKSDB_CHECK), 0)
 	TAGS += rocksdb
-endif 
+    CGO_CXXFLAGS := "${CGO_CXXFLAGS} -std=c++11"
+    CGO_LDFLAGS := "${CGO_LDFLAGS} -g -O2 -lrocksdb -lstdc++ -lm -lz -lbz2 -lsnappy -llz4 -lzstd"
```
Comment (siddontang) [inline Makefile]: can we remove `-lrocksdb -lm -lz -lbz2 -lsnappy -llz4 -lzstd` here? 

above check has already checked `-lrocksdb` and for the following links, I prefer passing outside the Makefile manually.

Comment (DorianZheng) [reply Makefile]: Yes, It seems that it's OK to remove it.

File: Gopkg.toml [hunk 3]
```diff
@@ -78,6 +78,7 @@
 [[constraint]]
   branch = "master"
   name = "github.com/tecbot/gorocksdb"
+  source = "https://github.com/DorianZheng/gorocksdb.git"
```
Comment (siddontang) [inline Gopkg.toml]: please update vendor too

**Pull request merged**